integrations/coinbase-integration/src/main/java/org/dash/wallet/integration/coinbase_integration/ui/CoinbaseConvertCryptoFragment.kt

```diff
@@ -183,47 +182,45 @@ class CoinbaseConvertCryptoFragment : Fragment(R.layout.fragment_coinbase_conver
         }
 
         binding.convertView.setOnCurrencyChooserClicked {
-            viewModel.getUserWalletAccounts(binding.convertView.dashToCrypto)
+            lifecycleScope.launch {
+                var list = listOf<CoinBaseUserAccountDataUIModel>()
+                val selectedCurrencyCode = convertViewModel.selectedCryptoCurrencyAccount.value
+                    ?.coinBaseUserAccountData?.currency?.code ?: convertViewModel.selectedLocalCurrencyCode
+
+                cryptoWalletsDialog = CryptoWalletsDialog(selectedCurrencyCode) { index, dialog ->
+                    list.getOrNull(index)?.let {
+                        convertViewModel.setSelectedCryptoCurrency(it)
+                        setConvertViewInput()
+                    }
+                    dialog.dismiss()
+                }
+
+                if (cryptoWalletsDialog?.isVisible == false) {
+                    viewModel.isDeviceConnectedToInternet.value?.let { hasInternet ->
+                        cryptoWalletsDialog?.handleNetworkState(hasInternet)
+                    }
+
+                    cryptoWalletsDialog?.show(parentFragmentManager, "payment_method")
+                    list = viewModel.getUserWalletAccounts(binding.convertView.dashToCrypto)
+
+                    if (list.isEmpty()) {
+                        showNoAssetsError()
+                    } else {
+                        cryptoWalletsDialog?.submitList(list)
+                    }
+                }
+            }
         }
 
         binding.convertView.setOnSwapClicked {
             convertViewModel.setOnSwapDashFromToCryptoClicked(it)
         }
 
-
         convertViewModel.selectedLocalExchangeRate.observe(viewLifecycleOwner) {
             binding.convertView.exchangeRate = ExchangeRate(Coin.COIN, it.fiat)
             setConvertViewInput()
         }
 
-        viewModel.userAccountsWithBalance.observe(
```

Comment:
Passing the execution flow around by calling/observing events is convoluted and unnecessary. Just call a suspend method from the fragment and wait for the result.